### DISCUSSION TITLE
How do you inject CSS (via stylus) to a certain DOM container?

### DISCUSSION THREAD

**Discussion by binarykitchen** [stylus/stylus#2897] [General]
Hello all, :)

Trying to migrate away from this deprecated npm package [insert-css](https://www.npmjs.com/package/insert-css) and still stuck.

I've got some CSS in stylus and would like to inject them to a certain DOM container like that deprecated package did. Used gulp for that, but now it's 2024. Usually, you let bundlers compile to CSS and link them to your website.

But what if you want to amend compiled CSS from Stylus to the HTML header at runtime? I haven't found a solution for that yet. LMK, thanks folks

**Comment by vendethiel** [comment]
do you need to compile the Stylus code dynamically? Wouldn’t it be simpler to just ship the CSS compiled ahead of time?

**Reply by binarykitchen** [reply]
Great question. 

Dynamically because it's a web component I intend to export as an independent npm package. And the parent HTML should be configurable. This deprecated npm package did it well ...

Ideas, suggestions welcome.
